spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/store/redis/JdkSerializationStrategy.java

```diff
@@ -1,26 +1,89 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package org.springframework.security.oauth2.provider.token.store.redis;
 
-import org.springframework.data.redis.serializer.JdkSerializationRedisSerializer;
+import org.springframework.core.serializer.support.SerializationFailedException;
+
+import java.io.Serializable;
+import java.util.List;
+import org.springframework.security.oauth2.common.util.SerializationStrategy;
 
 /**
- * Serializes objects using {@link JdkSerializationRedisSerializer}
+ * Serializes and deserializes allowed objects using {@link SerializationStrategy}
+ * which allows specifying a list of allowed classes for deserialization.
  *
  * @author efenderbosch
- *
+ * @author Artem Smotrakov
  */
 public class JdkSerializationStrategy extends StandardStringSerializationStrategy {
 
-	private static final JdkSerializationRedisSerializer OBJECT_SERIALIZER = new JdkSerializationRedisSerializer();
+  private static final byte[] EMPTY_ARRAY = new byte[0];
+
+  /**
+   * The SerializationStrategy to be used.
+   */
+  private final SerializationStrategy serializationStrategy;
+
+  /**
+   * The default constructor initializes a {@link JdkSerializationStrategy}
+   * with a default {@link SerializationStrategy};
+   */
+  public JdkSerializationStrategy() {
+    this.serializationStrategy = new SerializationStrategy();
+  }
+
+  /**
+   * Initializes a {@link JdkSerializationStrategy} with a list of allowed classes.
+   *
+   * @param allowedClasses The allowed classes.
+   */
+  public JdkSerializationStrategy(List<String> allowedClasses) {
```

Comment:
Please remove this constructor and add `setSerializationStrategy(SerializationStrategy serializationStrategy)`, now that `SerializationStrategy`  is an interface

Reply:
Okay, I can do that. I thought it might be better to make `serializationStrategy` filed final.